### ISSUE TITLE
Monitors: In case of unset filename variable, fallback to component instance name

### ISSUE THREAD

**Issue by willend** [McStasMcXtrace/McCode#1030]
Simple trick that will guarantee file-output even if the user did not specify a filename.

Good idea or bad? Comments welcome...

**Comment by farhi** [comment]
Should inspire from e.g. monitor_nD that does exactly this.

**Comment by willend** [comment]
Most of our comps default by having a setting pararameter `string filename=0` and resulting e.g. `_D0_Source_var._parameters.filename[0]='\0';` in the setpos routine

It turns out that this line at the start / end of `INITIALIZE` enables the desired behaviour:
```  if (!strcmp(filename,"\0")) sprintf(filename,NAME_CURRENT_COMP);```

**Comment by willend** [comment]
For now, I am defining this in the McStas `PSD_monitor` on McCode 3, but it is in principle a simple operation to do this across the monitors.

**Comment by willend** [comment]
Awaiting nightly tests for potential side-effects, but mccode-3 is now "ready for release" wrt. this issue.

**Comment by farhi** [comment]
Still there...

**Comment by willend** [comment]
@farhi yes, may in fact be there for McXtrace but was fixed for McStas. Keeping it open.

**Comment by farhi** [comment]
Fixed in McXtrace
